website/views.py

```diff
@@ -156,28 +156,30 @@ def profile_edit(request):
 def add_domain_to_company(request):
     if request.method == "POST":
         domain = request.POST.get("domain")
-        domain = Domain.objects.get(id=domain)
+        try:
+            domain = Domain.objects.get(id=domain)
+        except Domain.DoesNotExist:
+            return HttpResponseBadRequest("Invalid domain ID")
         company_name = request.POST.get("company")
         company = Company.objects.filter(name=company_name).first()
 
-        if not company:
+        try:
```

Comment:
This is very basic check for company name in the response body. Let's at least check that the request isn't being redirected to another location (status code 200 might work)

Reply:
I have made the changes, plz do check if this satisfies the requirements. Thank you.